src/java.base/share/classes/java/lang/foreign/FunctionDescriptor.java

```diff
@@ -81,8 +80,7 @@ public List<MemoryLayout> argumentLayouts() {
      */
     public static FunctionDescriptor of(MemoryLayout resLayout, MemoryLayout... argLayouts) {
         Objects.requireNonNull(resLayout);
-        Objects.requireNonNull(argLayouts);
-        Arrays.stream(argLayouts).forEach(Objects::requireNonNull);
+        assertVarargsNonNull(argLayouts);
```

Comment:
This check is redundant, since it is performed by `List.of` (granted that the stack trace is different but i think that is ok for NPEs).

Reply:
Well spotted!